c_glib/arrow-glib/basic-array.cpp

```diff
@@ -2806,6 +2822,265 @@ garrow_time64_array_get_values(GArrowTime64Array *array,
+ * @array: A #GArrowMonthIntervalArray.
+ * @length: (out): The number of values.
+ *
+ * Returns: (array length=length): The raw values.
+ *
+ * Since: 8.0.0
+ */
+const gint32 *
+garrow_month_interval_array_get_values(GArrowMonthIntervalArray *array,
+                                       gint64 *length)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  return garrow_array_get_values_raw<arrow::MonthIntervalType>(arrow_array, length);
+}
+
+
+G_DEFINE_TYPE(GArrowDayTimeIntervalArray,
+              garrow_day_time_interval_array,
+              GARROW_TYPE_NUMERIC_ARRAY)
+
+static void
+garrow_day_time_interval_array_init(GArrowDayTimeIntervalArray *object)
+{
+}
+
+static void
+garrow_day_time_interval_array_class_init(GArrowDayTimeIntervalArrayClass *klass)
+{
+}
+
+/**
+ * garrow_day_time_interval_array_new:
+ * @length: The number of elements.
+ * @data: The binary data in Arrow format of the array.
+ * @null_bitmap: (nullable): The bitmap that shows null elements. The
+ *   N-th element is null when the N-th bit is 0, not null otherwise.
+ *   If the array has no null elements, the bitmap must be %NULL and
+ *   @n_nulls is 0.
+ * @n_nulls: The number of null elements. If -1 is specified, the
+ *   number of nulls are computed from @null_bitmap.
+ *
+ * Returns: A newly created #GArrowDayTimeIntervalArray.
+ *
+ * Since: 8.0.0
+ */
+GArrowDayTimeIntervalArray *
+garrow_day_time_interval_array_new(gint64 length,
+                                   GArrowBuffer *data,
+                                   GArrowBuffer *null_bitmap,
+                                   gint64 n_nulls)
+{
+  auto array = garrow_primitive_array_new<arrow::DayTimeIntervalType>(length,
+                                                                      data,
+                                                                      null_bitmap,
+                                                                      n_nulls);
+  return GARROW_DAY_TIME_INTERVAL_ARRAY(array);
+}
+
+/**
+ * garrow_day_time_interval_array_get_value:
+ * @array: A #GArrowDayTimeIntervalArray.
+ * @i: The index of the target value.
+ *
+ * Returns: (transfer full): The @i-th value.
+ *
+ * Since: 8.0.0
+ */
+GArrowDayMillisecond *
+garrow_day_time_interval_array_get_value(GArrowDayTimeIntervalArray *array,
+                                         gint64 i)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  auto arrow_day_time_interval_array =
+    std::static_pointer_cast<arrow::DayTimeIntervalArray>(arrow_array);
+  auto arrow_day_time_interval = arrow_day_time_interval_array->GetValue(i);
+  return garrow_day_millisecond_new_raw(&arrow_day_time_interval);
+}
+
+/**
+ * garrow_day_time_interval_array_get_values:
+ * @array: A #GArrowDayTimeIntervalArray.
+ *
+ * Returns: (nullable) (element-type GArrowDayMillisecond) (transfer full):
+ *   The returned list of #GArrowDayMillisecond on success, %NULL on error.
```

Comment:
`NULL` isn't an error. If the number of elements is `0`, `NULL` is returned as a valid value. 

```suggestion
 *   The list of #GArrowDayMillisecond.
```